datalad/cmd.py

```diff
@@ -98,7 +98,208 @@ def _cleanup_output(stream, std):
+                     exc_str(e)))
+            raise
+
+        try:
+            out = [b'', b'']
+            last_idx = [0, 0]
+            pout = None
+            # make sure to run this loop at least once, even if the
+            # process is already dead
+            keep_going = True
+            while process.poll() is None or keep_going:
+                try:
+                    # get a chunk of output for the specific period
+                    # of time
+                    pout = process.communicate(
+                        timeout=self._poll_period,
+                    )
+                    # we know the process ended at this point
+                    keep_going = False
+                except subprocess.TimeoutExpired as poll:
+                    # this will always be the full report so far,
+                    # not just an output increment
+                    pout = (poll.stdout, poll.stderr)
+
+                if proc_out is not None:
+                    for i, (o, proc) in enumerate(zip(pout, proc_out)):
+                        if proc is None:
+                            if o is not None:
+                                out[i] += o
+                            # no index update needed, can never change
+                            continue
+                        # current length of output
+                        len_o = len(o) if o else 0
+                        # anything new in the output?
+                        if len_o - last_idx[i] > 0:
+                            # engage output processed
+                            processed, unprocessed_len = proc(o[last_idx[i]:])
+                            # only keep the processed output
+                            out[i] += processed
+                            # record point to which we processed the output
+                            # subtract number of bytes reported as unprocessed
+                            last_idx[i] = len_o - unprocessed_len
+
+            # obtain exit code
+            status = process.poll()
+
+            # decode bytes to string
+            out = tuple(o.decode('utf-8') if o else '' for o in out)
+
+            if status not in [0, None]:
+                msg = "Failed to run %r%s." % (
+                    cmd,
+                    (" under %r" % self.cwd) if self.cwd else '',
+                )
+                raise CommandError(
+                    cmd=str(cmd),
+                    msg=msg,
+                    code=status,
+                    stdout=out[0],
+                    stderr=out[1],
+                )
+            else:
+                lgr.log(8, "Finished running %r with status %s", cmd, status)
+
+        except CommandError:
+            # do not bother with reacting to "regular" CommandError
+            # exceptions.  Somehow if we also terminate here for them
+            # some processes elsewhere might stall:
+            # see https://github.com/datalad/datalad/pull/3794
+            raise
```

Comment:
I know this is copy-and-paste, but why catch at all here?

Reply:
The specific comment by @yarikoptic was https://github.com/datalad/datalad/pull/3794#issuecomment-544213473

I don't know any additional details. We can take it out and see if it has any effect.

Reply:
Doh, I misread the code and didn't realize that it was avoiding going into the `except BaseException` clause below it.  Sorry for the noise.

Reply:
Took it out.